adapter/oam.go

```diff
@@ -145,3 +157,74 @@ func (or *OAMRegistrant) Register() error {
 
 	return nil
 }
+
+type DynamicComponentsConfig struct {
+	TimeoutInMinutes time.Duration
+	URL              string
+	GenerationMethod string
+	Config           manifests.Config
+	Operation        string
+}
+
+func RegisterWorkLoadsDynamically(runtime, host string, dc *DynamicComponentsConfig) error {
+	var comp *manifests.Component
+	var err error
+	switch dc.GenerationMethod {
+	case Manifests:
+		comp, err = manifests.GetFromManifest(dc.URL, manifests.SERVICE_MESH, dc.Config)
+	case HelmCHARTS:
+		comp, err = manifests.GetFromHelm(dc.URL, manifests.SERVICE_MESH, dc.Config)
+	default:
+		return ErrGenerateComponents(errors.New("failed to generate components"))
+	}
+	if err != nil {
+		return ErrGenerateComponents(err)
+	}
+	for i, def := range comp.Definitions {
+		var ord OAMRegistrantData
+		ord.OAMRefSchema = comp.Schemas[i]
+
+		//Marshalling the stringified json
+		ord.Host = host
+		definitionMap := map[string]interface{}{}
+		if err := json.Unmarshal([]byte(def), &definitionMap); err != nil {
+			return err
+		}
+		definitionMap["apiVersion"] = "core.oam.dev/v1alpha1"
+		definitionMap["kind"] = "WorkloadDefinition"
+		ord.OAMDefinition = definitionMap
+		ord.Metadata = map[string]string{
+			OAMAdapterNameMetadataKey: dc.Operation,
+		}
+		// send request to the register
+		backoffOpt := backoff.NewExponentialBackOff()
+		backoffOpt.MaxElapsedTime = 10 * dc.TimeoutInMinutes
```

Comment:
Not sure but shouldn't this be `dc.TimeoutInMinutes * time.Minute`?

Reply:
yups,typo. Fixed it.